app/controllers/joins_controller.rb

```diff
@@ -0,0 +1,28 @@
+class JoinsController < ApplicationController
+  def create
+    @join = current_user.joins.build(join_params)
+    @recruit = Recruit.find_by(id: params[:recruit_id])
+    if current_user.id == @recruit.user_id
+      return
+    elsif current_user.already_commented?(@recruit) && current_user.id != @recruit.user_id
+      @join.save
+      flash[:notice] = 'You joined this recruit!'
+    else
+      flash[:alert] = 'Please comment first'
+    end
+
+    redirect_back(fallback_location: root_path)
+  end
+
+  def destroy
+    @join = current_user.joins.find_by(recruit_id: params[:recruit_id])
+    @join.destroy!
```

Comment:
find_byは見つからなかった場合にnil返すのでぼっち演算子あった方がいいかも。ここでnil undefindで落ちるのは不本意だと思うので。

```suggestion
    @join&.destroy!
```

Reply:
了解しました。変更しました！